src/envs/pddl_env.py

```diff
@@ -0,0 +1,442 @@
+
+
+###############################################################################
+#                              Concrete Envs                                  #
+###############################################################################
+
+
+class _BlocksPDDLEnv(_PDDLEnv):
+    """The IPC 4-operator blocks world domain."""
+
+    @property
+    def _domain_str(self) -> str:
+        path = utils.get_env_asset_path("pddl/blocks/domain.pddl")
+        with open(path, encoding="utf-8") as f:
+            domain_str = f.read()
+        return domain_str
+
+
+class FixedTasksBlocksPDDLEnv(_BlocksPDDLEnv, _FixedTasksPDDLEnv):
+    """The IPC 4-operator blocks world domain with a fixed set of tasks."""
+
+    @classmethod
+    def get_name(cls) -> str:
+        return "pddl_blocks_fixed_tasks"
+
+    @property
+    def _pddl_problem_asset_dir(self) -> str:
+        return "blocks"
+
+    @property
+    def _train_problem_indices(self) -> List[int]:
+        return list(range(1, 6))
+
+    @property
+    def _test_problem_indices(self) -> List[int]:
+        return list(range(6, 11))
+
+
+class ProceduralTasksBlocksPDDLEnv(_BlocksPDDLEnv):
+    """The IPC 4-operator blocks world domain with procedural generation."""
+
+    @classmethod
+    def get_name(cls) -> str:
+        return "pddl_blocks_procedural_tasks"
+
+    @property
+    def _pddl_train_problem_generator(self) -> PDDLProblemGenerator:
+        min_num_blocks = CFG.pddl_blocks_train_min_num_blocks
+        max_num_blocks = CFG.pddl_blocks_train_max_num_blocks
+        min_num_blocks_goal = CFG.pddl_blocks_train_min_num_blocks_goal
+        max_num_blocks_goal = CFG.pddl_blocks_train_max_num_blocks_goal
+        new_pile_prob = CFG.pddl_blocks_new_pile_prob
+        return create_blocks_pddl_generator(min_num_blocks, max_num_blocks,
+                                            min_num_blocks_goal,
+                                            max_num_blocks_goal, new_pile_prob)
+
+    @property
+    def _pddl_test_problem_generator(self) -> PDDLProblemGenerator:
+        min_num_blocks = CFG.pddl_blocks_test_min_num_blocks
+        max_num_blocks = CFG.pddl_blocks_test_max_num_blocks
+        min_num_blocks_goal = CFG.pddl_blocks_test_min_num_blocks_goal
+        max_num_blocks_goal = CFG.pddl_blocks_test_max_num_blocks_goal
+        new_pile_prob = CFG.pddl_blocks_new_pile_prob
+        return create_blocks_pddl_generator(min_num_blocks, max_num_blocks,
+                                            min_num_blocks_goal,
+                                            max_num_blocks_goal, new_pile_prob)
+
+
+###############################################################################
+#                            Utility functions                                #
+###############################################################################
+
+
+def _state_to_ground_atoms(state: State) -> Set[GroundAtom]:
+    return cast(PDDLEnvSimulatorState, state.simulator_state)
```

Comment:
can we remove this PDDLEnvSimulatorState thing entirely? i think it's more confusing than helpful